location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,466 @@
+       video_id: The video_id of specific audio file from which features are to
+           be extracted.
+       vid_id_to_features: A dictionary with video_id, dictionary pairs with the
+           dictionary values being feature name, feature list key, value pairs.
+
+   Returns:
+       A boolean of whether the extracting features was successful
+   """
+    feature_dict = {}
+    if not isfile(FLAGS.dest_dir + '/yt_videos/sliced_' + video_id + '.wav'):
+        return False
+    y, sr = librosa.load(FLAGS.dest_dir + '/yt_videos/sliced_'
+                         + video_id + '.wav')
+    feature_dict["chroma_stft"] = librosa.feature.chroma_stft(y, sr)
+    feature_dict["chroma_cqt"] = librosa.feature.chroma_cqt(y, sr)
+    feature_dict["chroma_cens"] = librosa.feature.chroma_cens(y, sr)
+    feature_dict["melspectrogram"] = librosa.feature.melspectrogram(y, sr)
+    feature_dict["mfcc"] = librosa.feature.mfcc(y, sr)
+    feature_dict["rms"] = librosa.feature.rms(y)
+    feature_dict["spectral_centroid"] = librosa.feature.spectral_centroid(y, sr)
+    feature_dict["spectral_bandwidth"] = librosa.feature.spectral_bandwidth(y,
+                                                                            sr)
+    feature_dict["spectral_contrast"] = librosa.feature.spectral_contrast(y, sr)
+    feature_dict["spectral_flatness"] = librosa.feature.spectral_flatness(y)
+    feature_dict["spectral_rolloff"] = librosa.feature.spectral_rolloff(y, sr)
+    feature_dict["poly_features"] = librosa.feature.poly_features(y, sr)
+    feature_dict["tonnetz"] = librosa.feature.tonnetz(y, sr)
+    feature_dict["zero_crossing_rate"] = librosa.feature.zero_crossing_rate(y,
+                                                                            sr)
+    vid_id_to_features[video_id] = feature_dict
+    logging.info("extracted features")
+    return True
+
+
+def create_csv_from_list(vid_id_to_features, labels_dict):
+    """Creates a csv from a dictionary of video_id, feature dictionary
+
+    Creates a csv file to input into TensorFlow with the following structure:
+    is_gunshot, video_id, feature list.
+
+    Args:
+        vid_id_to_features: A dictionary with video_id, dictionary pairs with
+            the dictionary values being feature name, feature list key,
+            value pairs.
+        labels_dict: a dictionary of video_id list of labels key value pairs
+    """
+    with open(FLAGS.dest_dir + "/tfdataset.csv", 'w') as outcsv:
+        csv_writer = csv.writer(outcsv, delimiter=':',
+                                quoting=csv.QUOTE_MINIMAL)
+        csv_writer.writerow(
+            ['label', 'video_id', 'chroma_stft', 'chroma_cqt', 'chroma_cens',
+             'melspectrogram', 'mfcc', 'rms', 'spectral_centroid',
+             'spectral_bandwidth', 'spectral_contrast', 'spectral_flatness',
+             'spectral_rolloff', 'poly_features', 'tonnetz',
+             'zero_crossing_rate'])
+        labels_set = label_list_to_set(FLAGS.labels)
```

Comment:
i guess you can just call set() here instead of making label_list_to_set as a function

Reply:
Ah. Ok. I'll make that change. Thank you.